main.tf

```diff
@@ -15,7 +15,7 @@ locals {
 }
 
 resource "aws_iam_policy" "this" {
-  count = length(var.iam_instance_profile_arn) == 0 ? 1 : 0
+  count = var.create_instance_profile ? 1 : 0
```

Comment:
Maybe we should rename 'create_instance_profile' to 'create_iam_resources' since it is also creating roles / policies and not just the instance profile.